### DISCUSSION TITLE
How are links parsed?

### DISCUSSION THREAD

**Discussion by KBHertzog** [Milkdown/milkdown#1525] [Q&A] [answered]
I'm working on adding Milkdown support for a project I work on and noticed the standard Markdown format of `[Display Text](link)` wasn't working. Using a listener to print the markdown as it's parsed, it seems instead links are wrapped in <> tags. What's the reason as to why the Commonmark specification for links doesn't seem to work?

**Comment by KBHertzog** [comment]
Noting that further research lead me to learn these are proper links known as [auto-links](https://spec.commonmark.org/0.31.2/#autolink) that I hadn't encountered prior. It then makes sense that the <> output is supported, but I am still confused as to why the `[]()` syntax doesn't seem to work within the editor.

**Comment by ryne6** [comment]
Hi, I encountered the same issue, have you made any progress?

**Comment by Saul-Mirone** [comment, marked as answer]
Basically you need to enable plugin-automd or write a prosemirror input rule to parse the link. I plan to add the input rule in the common mark preset.
